great_expectations/data_context/store/store.py

```diff
@@ -98,7 +98,9 @@ def _validate_key(self, key: DataContextKey) -> None:
         # STORE_BACKEND_ID_KEY always validated
         if key == StoreBackend.STORE_BACKEND_ID_KEY:
             return
-        elif not isinstance(key, self.key_class):
+        elif isinstance(key, self.key_class):
```

Comment:
This is the same logic as before right just a little cleaned up?
You could also do this or just removed the last `else` to remove a branch:
```
if key == StoreBackend.STORE_BACKEND_ID_KEY or isinstance(key, self.key_class):
    return
raise TypeError(
                f"key must be an instance of {self.key_class.__name__}, not {type(key)}"
            )
```
Also, the raised `TypeError` seems a little wrong since `StoreBackend.STORE_BACKEND_ID_KEY` is also valid but maybe the user doesn't care about that?

Reply:
Yes, the logic is the same. I prefer the `else` because I find it more readable, and I doubt the compiler treats it any differently.

The user doesn't know about `StoreBackend.STORE_BACKEND_ID_KEY`. It references a hidden file (`.ge_store_backend_id`) in the `FileDataContext` `ExpectationsStore`.